src/OpenTelemetry.Api.ProviderBuilderExtensions/Logs/OpenTelemetryDependencyInjectionLoggerProviderBuilderExtensions.cs

```diff
@@ -14,6 +14,9 @@
 // limitations under the License.
 // </copyright>
 
+#if NET6_0_OR_GREATER
```

Comment:
We have UnconditionalSuppressMessageAttribute shim checked into the repo:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/a808276efaf7e7a13420d993cea903cbff262996/src/Shared/Shims/UnconditionalSuppressMessageAttribute.cs#L30

Should we continue down that path here? Using both approaches feels wrong.

Reply:
The team decided to unify TFMs across the projects and include `net6.0` in all of them and then use ifdefs to only add the annotations to the `net6.0` versions.
So to make things consistent we should probably remove the shim for `UnconditionalSuppressMessageAttribute` and use ifdefs there as well.

@Yun-Ting as fyi

Reply:
> The team decided to unify TFMs across the projects and include `net6.0` in all of them and then use ifdefs to only add the annotations to the `net6.0` versions. So to make things consistent we should probably remove the shim for `UnconditionalSuppressMessageAttribute` and use ifdefs there as well.
> 
I Agree. Please see the issue: https://github.com/open-telemetry/opentelemetry-dotnet/issues/4591

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4696